www/src/pages/en/recipes/include-projection-gsi.mdx

```diff
@@ -0,0 +1,290 @@
+
+### Basic Querying
+
+```typescript
+// Only returns the projected attributes: name, status, createdAt
+const { data, cursor } = await tasks.query
+  .statusIndex({ status: "open" })
+  .go();
+
+// data will only contain: { name: string, status: string, createdAt: number }[]
+```
+
+### Filtering on Projected Attributes
+
+You can only filter on attributes that are included in the projection:
+
+```typescript
+// ✅ This works - filtering on projected attributes
+const { data } = await tasks.query
+  .statusIndex({ status: "open" })
+  .where((attr, op) => op.eq(attr.name, "Important Task"))
+  .go();
+
+const { data } = await tasks.query
+  .statusIndex({ status: "open" })
+// ❌ This would not work - filtering on non-projected attributes
+  .where((attr, op) => op.eq(attr.description, "Some description"))
+  .go();
+```
+
+### Using Hydration
+
+If you need access to non-projected attributes, you can use the `hydrate` option to perform a follow-up `batchGet` operation:
+
+```typescript
+// With hydration, you get all attributes but can only filter on projected ones
+const { data } = await tasks.query
+  .statusIndex({ status: "open" })
+  .where((attr, op) => op.eq(attr.name, "Important Task"))
+  .go({ hydrate: true });
+
+// data will contain all attributes: { taskId, projectId, name, description, status, priority, createdAt, updatedAt }[]
+```
+
+### Selective Attribute Retrieval with Hydration
+
+When using hydration, you can still specify which attributes to return:
+
+```typescript
+const { data } = await tasks.query
+  .statusIndex({ status: "open" })
+  .where((attr, op) => op.eq(attr.name, "Important Task"))
+  .go({ 
+    hydrate: true, 
+    attributes: ['taskId', 'name', 'description'] 
+  });
+
+// data will only contain: { taskId: string, name: string, description: string }[]
+```
+
+## Benefits of INCLUDE Projections
+
+1. **Cost Optimization**: Only store the attributes you need in the index
+2. **Performance**: Smaller index size means faster queries
+3. **Multi-Filter Efficiency**: Create compact indexes for complex filtering patterns. When you have access patterns requiring many optional filters, projecting only necessary attributes creates a smaller "virtual table" that enables efficient multi-filter queries without scanning unnecessary data
+4. **Aggregation Optimization**: Smaller projected indexes are ideal for aggregation queries, reducing the number of requests needed and lowering costs by not scanning irrelevant attributes
+5. **Type Safety**: ElectroDB enforces which attributes are available at the type level
+
+## ⚠️ Critical Considerations
+
+### Entity Identifier Requirements
+
+When using `INCLUDE` projections, you **must** include ElectroDB's internal entity identifier attributes (`__edb_e__` and `__edb_v__`) in your projection. These attributes are essential for ElectroDB's multi-entity guardrail checks.
```

Comment:
I can own this, but we'll want to create an abstraction for this somehow. I'm still soul searching on the best approach (and open to ideas), but I typically try to make sure things "just work" (™) for greenfield devs. 

For a few reasons, my instinct is to disable ownership checks by default and use opt-in for developers who include identifiers in their GSI projection attributes. 

The most significant exposure here is for Services that use a collection on an `INCLUDE` GSI that also uses the `template` syntax. The move here might be to recognize the risky scenario and branch into an additional validation at Service instantiation; namely, more stringent checks to ensure isolation in key formatting.

These are just musings right now, though I'm interested in your thoughts.